packages/ember-glimmer/lib/environment.js

```diff
@@ -96,6 +100,7 @@ export default class Environment extends GlimmerEnvironment {
 
     let nativeSyntax = super.refineStatement(statement);
     assert(`Helpers may not be used in the block form, for example {{#${key}}}{{/${key}}}. Please use a component, or alternatively use the helper in combination with a built-in Ember helper, for example {{#if (${key})}}{{/if}}.`, !nativeSyntax && key && this.hasHelper(key) ? !isBlock : true);
+    assert(`Helpers may not be used in the element form.`, !nativeSyntax && key && this.hasHelper(key) ? !isModifier : true);
```

Comment:
Can you include the example in here too? i.e. https://github.com/emberjs/ember.js/blob/master/packages/ember-htmlbars/lib/system/invoke-helper.js#L13